ADR/0032-decoupling-deployment.md

```diff
@@ -0,0 +1,223 @@
+# 32. Decoupling Deployment
+
+Date: 2023-11-17
+
+## Status
+
+Draft
+
+Relates to:
+
+- [ADR 14. Let Pipelines Proceed](0014-let-pipelines-proceed.html)
+
+Supercedes:
+
+- [ADR 08. Environment Provisioning](0008-environment-provisioning.html)
+- [ADR 16. Integration Service Promotion Logic](0016-integration-service-promotion-logic.html)
+- [ADR 28. Handling SnapshotEnvironmentBinding Errors](0028-handling-snapshotenvironmentbinding-errors.html)
+
+## Authors
+
+- Ralph Bean
+
+## Context
+
+Since the beginning of our project, we've had an emphasis on providing an integrated experience for
+the user, automating all steps through build, test, deployment, and release to higher environments.
+
+Some challenges:
+
+- Our controllers' APIs are highly coupled. For example, it isn't feasible to try and use
+  [integration-service] today without [application-service] and [gitops-service]. This creates
+  a high barrier to engage with some of the useful subsystems we've built. A user (and potential
+  contributor) has to adopt the whole thing, or nothing.
+- Today, we can generate simple but often incorrect deployment manifests for users' applications,
+  which we deploy via the application gitops repo. The user cannot interact with or influence these
+  resources directly; they have to do so through the Application/Component API. If we permit them to
+  directly provide deployment resources through that API, the API will become a leaky abstraction.
+  If don't permit them to provide their own resources, and instead try to handle every application
+  configuration case in the HAS generation code, we will struggle to keep up with users' demands for
+  the many and varied kinds of application deployments.
+
+See also [RHTAP-1873](https://issues.redhat.com/browse/RHTAP-1873).
+
+## Decision
+
+We are going to decouple the deployment from the build, test, and release portions of our system.
+
+**Deployment**:
+
+- The [Environment], [SnapshotEnvironmentBinding], and [GitOpsDeploymentManagedEnvironment]
+  resources will be deprecated and eventually decomissioned.
+- The [application-service] will stop generating GitOps repo content and stop creating GitOps
+  repositories.
+- We will stop deploying the [gitops-service] entirely and the RDS instance it uses.
+- HAC will stop rendering [Environments] and their status.
+- If a user wants to make use of deployment capabilities, we will promote the usage of [renovatebot]
+  to propagate _released_ images to their self-managed gitops repo as pull requests. Dependabot is
+  equally viable if the user's gitops repo is on GitHub.
+
+**Test and promotion**:
+
+- We will retain the [DeploymentTargetClaim], [DeploymentTarget], and [DeploymentTargetClass] APIs.
```

Comment:
@ralphbean based on the discussion we had last week, we can drop this CRD's as well. In addition, the integration service won't be responsible for requesting test environments. Would you like me to update this ADR?

Reply:
Agreed! I've got a draft update started locally. Didn't finish on Friday. I'll have it posted here before end of day.

Reply:
Updated in 9acabdcc0e632c13c2b8772d9ac8e75597e2b725.